csv_generator/process_xml_zip.py

```diff
@@ -0,0 +1,95 @@
+import argparse
+import os
+import shutil
+from typing import List
+import zipfile
+
+from bs4 import BeautifulSoup
+
+from consumers import feed_consumers, make_soup
+from consumers.utils import timestamp_to_epoch
+
+
+DEFAULT_ZIP_OUTPUT_DIR = '../zip_output'
+DEFAULT_OUTPUT_DIR = '../output'
+GO_XML_FILE_NAME = 'go.xml'
+
+
+def clean_up_zip(target_dir: str) -> None:
+    """Delete target zip output directory.
+
+    :param target_dir: str
+    :return:
+    """
+    try:
+        shutil.rmtree(target_dir)
+    except FileNotFoundError:
+        pass
+
+
+def extract_zip_file(target_zip: str, output_dir: str) -> None:
+    """Extract `target_zip` to `output_dir`
+
+    :param target_zip: str
+    :param output_dir: str
+    :return:
+    """
+    zip_file = zipfile.ZipFile(target_zip)
+
+    if not os.path.isdir(output_dir):
+        os.mkdir(output_dir)
+
+    zip_file.extractall(path=output_dir)
+
+
+def get_create_date(soup: BeautifulSoup) -> str:
+    """Return `create_date` from soup.
+
+    :param soup:
+    :return: class: `BeautifulSoup`
+    """
+    return soup.find('file_list')['create_date']
+
+
+def get_file_list(soup: BeautifulSoup) -> List[str]:
+    """Return `file_nm` from soup.
+
+    :param soup:
+    :return: list
+    """
+    return [name.contents[0] for name in soup.find_all('file_nm')]
+
+
+def main():
+    parser = argparse.ArgumentParser(description='process_zip')
+
+    parser.add_argument('--target-zip', type=str, help='Target zip file e.g. some-file.zip')
+    parser.add_argument('--output-dir', default=DEFAULT_OUTPUT_DIR,
+                        type=str, help='CSV output directory. Defaults to {}'.format(DEFAULT_OUTPUT_DIR))
+
+    args = parser.parse_args()
+
+    target_zip = args.target_zip
+    output_dir = args.output_dir
+
+    print('processing: ', target_zip)
+
+    extract_zip_file(target_zip=target_zip, output_dir=DEFAULT_ZIP_OUTPUT_DIR)
```

Comment:
What would you think of not extracting the files to disk but instead feeding the streams to the consumer instead? (or in this case it's passing the parsed XML file)

Reply:
I tried to keep it simple enough to just require a list of file paths to be passed to the `fedd_consumers` function, so it made sense to just unzip to disk before hand. I also needed to parse the `go.xml` file first in this instance before doing the initializing the consumers as we needed `create_date` from it.

I like the idea, though as I am already only parsing each file once then passing the `soup` object around there is already minimal io. I think its ok for now.